service/matching/matcher_test.go

```diff
@@ -406,8 +406,11 @@ func (t *MatcherTestSuite) TestMustOfferRemoteMatch() {
 		},
 	).Return(&matchingservice.AddWorkflowTaskResponse{}, nil)
 
+	time.Sleep(time.Second)
```

Comment:
note: this was causing flakiness on my box until I added a sleep. Unlike the 2nd flakiness fix down below, this is a bit more suspicious

Reply:
resolved this. race condition in the test. added a comment explaining why the sleep makes things better